text/0854-ember-data-cache-v2.1.md

```diff
@@ -0,0 +1,1478 @@
+
+Need to edit a lot of relational data and save it in one go? Use a fork! If you later
+want to discard the edits, instead of trying to keep track of what changed, just toss
+away the fork. Or, if you want to persist it, save all changes to the fork in one single
+go.
+
+Want to have some data loaded for all routes, but have other routes or engines manage their
+own data but still correctly be able to reference the global data? That's right, use forks!
+
+Want to be able to edit a whole lot of data and serialize the changes for local storage for
+a save/restore feature for a form, but don't want to pollute those records for where they
+appear elsewhere in your UI? Use a fork!
+
+Is it a bird, is it a plane? It might sound like a knife and feed you as easily as a spoon...
+but its a fork!
+
+### The Grand Vision (Abbreviated)
+
+This RFC codifies the direction and vision that the EmberData team has been building
+towards in our long-term strategy since early in my tenure on the team. To anyone who
+has been in these discussions the ideas presented here are not new, and we already know
+how they fit neatly into that vision; but for the community at large many of these
+changes may still be a surprise.
+
+This RFC is not "the grand vision", but it is an integral piece. Without it, changes we
+have planned for the presentation, network and store primitives will not be possible.
+
+Above, I have laid out several key use-cases and motivations for the changes herein,
+but the observant will recognize that this RFC does not expose APIs or specify how
+the store, network, or current model primitives might make use of these cache APIs. We
+leave each of these to their own RFCs – coming very soon – but for the sake of adequate
+discussion herein we outline the direction we expect those RFCs to take.
+
+**The APIs shown in this section are not a proposed design but representative of the overall vision for EmberData in the near future** 
+
+**`From Fetch to Fully Distributed Replica Database`**
+
+For the network layer, we see the introduction of `FetchManager`, a managed flow
+for making requests that allows registration of middleware to handle fulfillment
+or decoration of the request or response. This service would thereby make headers
+and auth management easy to handle, letting apps that previously had multiple
+unorganized services or ad-hoc fetch handlers to unify over one conventional flow.
+
+This managed `Fetch` flow would (by default) not be integrated with the Store, thereby
+skipping the cache and providing access to raw responses.
+
+```ts
+class FetchManager {
+  registerMiddlewares(wares: Middleware[]);
+
+  async request(req: RequestOptions): RequestFuture;
+}
+```
+
+For the store, we see the addition of a new API – `Store.request` – overwhich all
+existing store methods will be rebuilt as "convenience macros". This delegates to
+the above `FetchService`, but passes the Store's cache (which is potentially a fork)
```

Comment:
```suggestion
the above `FetchManager`, but passes the Store's cache (which is potentially a fork)
```

should this section now mention #860 where this part is explain in great detail?

Reply:
Yeah, I need to update the interfaces slightly too for RequestInfo ResponseInfo and StructuredDocument